### PR TITLE
Update Visual Studio SDK to 16.9

### LINKED ISSUE #10297: VS SDK:  update to 16.9.x
The following work items are blocked on updating the NuGet.Client solution to VS SDK 16.9.x:

*  https://github.com/NuGet/Home/issues/9935
*  https://github.com/NuGet/Home/issues/10167

I believe this task is at least partially blocked by [@heng-liu 's task](https://github.com/NuGet/Client.Engineering/issues/367) to update MSBuild dependencies and CI machines to 16.8.x.

### PR DESCRIPTION
**Pull request by dtivel**
## Bug

Fixes: https://github.com/nuget/home/issues/10297
Regression: No

## Fix

Details: Update the Visual Studio SDK to 16.9.x.  Because Microsoft.VisualStudio.SDK is a metapackage, it offers an opportunity to clean up and simplify our package references.

This PR contains some changes which are not strictly package reference updates but were necessary to build.

*  An updated analyzer flagged a bunch of places which needed assertions that the current thread is the UI thread.
*  Some VS credential provider tests had to be removed because they newly raised the underdocumented CS1769:

Severity | Code | Description | Project | File | Line | Source | Suppression State
-- | -- | -- | -- | -- | -- | -- | --
Error | CS1769 | Type 'Lazy<IVsCredentialProvider>' from assembly   'NuGet.PackageManagement.VisualStudio, Version=5.9.0.0, Culture=neutral,   PublicKeyToken=b03f5f7f11d50a3a' cannot be used across assembly boundaries   because it has a generic type argument that is an embedded interop type. | NuGet.PackageManagement.VisualStudio.Test | E:\git\NuGet.Client.3\test\NuGet.Clients.Tests\NuGet.PackageManagement.VisualStudio.Test\VsCredentialProviderImporterTests.cs | 120 | IntelliSense | Active
Error | CS1769 | Type 'Lazy<IVsCredentialProvider>' from assembly   'NuGet.PackageManagement.VisualStudio, Version=5.9.0.0, Culture=neutral,   PublicKeyToken=b03f5f7f11d50a3a' cannot be used across assembly boundaries   because it has a generic type argument that is an embedded interop type. | NuGet.PackageManagement.VisualStudio.Test | E:\git\NuGet.Client.3\test\NuGet.Clients.Tests\NuGet.PackageManagement.VisualStudio.Test\VsCredentialProviderImporterTests.cs | 89 | IntelliSense | Active
Error | CS1769 | Type 'Lazy<IVsCredentialProvider>' from assembly   'NuGet.PackageManagement.VisualStudio, Version=5.9.0.0, Culture=neutral,   PublicKeyToken=b03f5f7f11d50a3a' cannot be used across assembly boundaries   because it has a generic type argument that is an embedded interop type. | NuGet.PackageManagement.VisualStudio.Test | E:\git\NuGet.Client.3\test\NuGet.Clients.Tests\NuGet.PackageManagement.VisualStudio.Test\VsCredentialProviderImporterTests.cs | 113 | IntelliSense | Active
Error | CS1769 | Type 'Lazy<IVsCredentialProvider>' from assembly   'NuGet.PackageManagement.VisualStudio, Version=5.9.0.0, Culture=neutral,   PublicKeyToken=b03f5f7f11d50a3a' cannot be used across assembly boundaries   because it has a generic type argument that is an embedded interop type. | NuGet.PackageManagement.VisualStudio.Test | E:\git\NuGet.Client.3\test\NuGet.Clients.Tests\NuGet.PackageManagement.VisualStudio.Test\VsCredentialProviderImporterTests.cs | 160 | IntelliSense | Active

## Testing/Validation

Tests Added: No  
Reason for not adding tests:  not necessary

### CODE REVIEW

File: build/packages.targets
Comment (nkolev92) [inline build/packages.targets]: I'm not sure I understand the comment. If they are transitive there should be no need to specify them right?
This is only necessary if they are specified.

File: build/packages.targets
Comment (nkolev92) [inline build/packages.targets]: I'd prefer if we kept these packages alphabetically.

File: src/NuGet.Clients/NuGet.Console/WpfConsole/WpfConsole.cs
Comment (nkolev92) [inline src/NuGet.Clients/NuGet.Console/WpfConsole/WpfConsole.cs:863]: Is the analyzer complain about this? 

The `Invoke` method itself switch on the UI thread.

File: test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/NuGet.PackageManagement.UI.Test.csproj
Comment (nkolev92) [inline test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/NuGet.PackageManagement.UI.Test.csproj:15]: Why do we need this?
Isn't `PrivateAssemblies` already on the search path? 
We do that with: https://github.com/NuGet/NuGet.Client/blob/7e23737fdf89fe34c90ac8878c96843b2f0dd17e/src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/NuGet.PackageManagement.VisualStudio.csproj#L21-L23, which also comes from the same folder.

File: test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/VsCredentialProviderImporterTests.cs
Comment (nkolev92) [inline test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/VsCredentialProviderImporterTests.cs:81]: You are probably hitting the compile problem because there's an assembly that's supposed to be embedded, but it's not getting embedded.

Usually it's something like: 

Proj A -> Proj B. 

Both depend on AssemblyX.dll which needs to be embedded. 
This can happen if ProjB doesn't embed, but ProjA embeds I think.
So usually the answer to that error would be to add the `embedinterop.targets` to a new project or add a new assembly to that list.

**Review by nkolev92 — COMMENTED** [review]
Looks great overall. Great work!

Couple of questions and a suggestion for fixing the compile error.

File: src/NuGet.Clients/NuGet.Console/WpfConsole/WpfConsole.cs
Comment (dtivel) [reply src/NuGet.Clients/NuGet.Console/WpfConsole/WpfConsole.cs:863]: Yes, the analyzer specifically complained about this.  I only addressed threading issues that the analyzer complained about.

File: test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/NuGet.PackageManagement.UI.Test.csproj
Comment (dtivel) [reply test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/NuGet.PackageManagement.UI.Test.csproj:15]: This is an existing reference that I ordered amongst other references.  I'll see if this can be removed, but it's technically outside the scope of the PR, and I won't hold up merging for this.

Comment (nkolev92) [reply test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/NuGet.PackageManagement.UI.Test.csproj:15]: I missed that it was already there. Not relevant then. Your call.

File: test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/VsCredentialProviderImporterTests.cs
Comment (dtivel) [reply test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/VsCredentialProviderImporterTests.cs:81]: I tried adding setting this property all around, and it doesn't work.  I also tried setting the property via reflection and it doesn't work.

NuGet.VisualStudio should be embedded by default anyway.

**Review by nkolev92 — APPROVED** [review]

**Review by zivkan — APPROVED** [review]

File: test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/VsCredentialProviderImporterTests.cs
Comment (dtivel) [reply test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/VsCredentialProviderImporterTests.cs:81]: With this change NuGet.PackageManagement.VisualStudio brings in package Microsoft.VisualStudio.SDK.EmbedInteropTypes, which brings in a build target that sets EmbedInteropTypes to true for NuGet.VisualStudio.  That's what I meant when I said "NuGet.VisualStudio should be embedded by default anyway."

Prior to this change, NuGet.PackageManagement.VisualStudio did not bring in that build target, so EmbedInteropTypes was false for NuGet.VisualStudio.  That's how the now deleted tests used to pass.

It may be possible to not embed interop types for NuGet.VisualStudio after this change, but the new default seems correct to me.

**Pull request merged**
